### PR TITLE
Add nn.CrossEntropyLoss Module

### PR DESCRIPTION
**Pull request by doombeaker**
- CrossEntropyLoss  Module
- Sub Module
- Mul Module
- Mean Module
- Sum Module
- Div Module
- Reciprocal Module

### CODE REVIEW

**Comment by doombeaker** [comment]
（备忘）用户角度待解决的问题：
1. 在 forward 中，可否直接使用已有的 op，而不是需要自己通过 builtin_op 构建
2. 有时 Attr 需要的是 forward 时才能拿到的数据，这样，op 的 Buid 没法在 `__init__` 中完成，而需要放在 forward 中（如这个PR的op），这种做法不知道是否合理，即使合理，貌似也有点不好看。

File: oneflow/python/nn/modules/loss.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            flow.builtin_op("sparse_softmax_cross_entropy")
+            .Name(_opname)
+            .Input("prediction")
+            .Input("label")
+            .Output("prob")
+            .Output("out")
+        )
+
+    def forward(self, input, target):
+        self._op = self._op.Attr("depth", input.shape[len(input.shape) - 1]).Build()
```
Comment (doombeaker) [inline oneflow/python/nn/modules/loss.py:62]: Tensor 的 shape 暂时还不支持负数的索引（如：-1）
